src/compiler/test-file/parse-es-next.js

```diff
@@ -0,0 +1,241 @@
+import { assign } from 'lodash';
+import { format } from 'util';
+import { transform as parse } from 'babel-core';
+import { GeneralError } from '../../errors/runtime';
+import ESNextTestFileCompiler from './formats/es-next';
+
+import promisify from '../../utils/promisify';
+import fs from 'fs';
+
+import MESSAGE from '../../errors/runtime/message';
+
+var readFile = promisify(fs.readFile);
+
+const COMPUTED_NAME_TEXT_TMP  = '<computed name>(line: %s)';
+const METHODS_SPECIFYING_NAME = ['only', 'skip'];
+
+function getRValue (token) {
+    return token.declarations[0].init;
+}
+
+function getFunctionBody (token) {
+    return token.body && token.body.body ? token.body.body : [];
+}
+
+function getTagStrValue (exp) {
+    //NOTE: we set <computed name> if template literal has at least one computed substring ${...}
+    return exp.expressions.length ? format(COMPUTED_NAME_TEXT_TMP, exp.loc.start.line) : exp.quasis[0].value.raw;
+}
+
+function formatFnData (name, value, token) {
+    return {
+        fnName: name,
+        value:  value,
+        loc:    token.loc,
+        start:  token.start,
+        end:    token.end
+    };
+}
+
+function checkExpDefineTargetName (type, apiFn) {
+    //NOTE: fixture('fixtureName') or test('testName')
+    const isDirectCall = type === 'Identifier';
+
+    //NOTE: fixture.skip('fixtureName'), test.only('testName') etc.
+    const isMemberCall = type === 'MemberExpression' && METHODS_SPECIFYING_NAME.indexOf(apiFn) > -1;
+
+    //NOTE: fixture.before().after()('fixtureName'), test.before()`testName`.after() etc.
+    const isTailCall = type === 'CallExpression';
+
+    return isDirectCall || isMemberCall || isTailCall;
+}
+
+function analyzeMemberExp (token) {
+    let exp         = token;
+    const callStack = [exp];
+
+    while (exp.type !== 'Identifier') {
+        if (exp.type === 'CallExpression')
+            exp = exp.callee;
+
+        else if (exp.type === 'MemberExpression')
+            exp = exp.object;
+
+        else if (exp.type === 'TaggedTemplateExpression')
+            exp = exp.tag;
+
+        if (exp.type !== 'Identifier')
+            callStack.push(exp);
+    }
+
+    if (exp.name !== 'fixture' && exp.name !== 'test')
+        return null;
+
+    let parentExp = callStack.pop();
+
+    if (parentExp.type === 'CallExpression')
+        return formatFnData(exp.name, formatFnArg(parentExp.arguments[0]), token);
+
+    if (parentExp.type === 'TaggedTemplateExpression')
+        return formatFnData(exp.name, getTagStrValue(parentExp.quasi), token);
+
+    if (parentExp.type === 'MemberExpression') {
+        /*eslint-disable no-cond-assign*/
```

Comment:
Is there a reason to ignore this rule? As I understand this rule is created for avoid such constructions like `while(field=value)`...